pkg/controllers/awsloadbalancercontroller/rbac_clusterrolebinding.go

```
@@ -0,0 +1,138 @@
+	reqLogger := log.FromContext(ctx)
+
+	desired := desiredClusterRoleBinding(ctx, sa)
+	reqLogger.Info("ensuring clusterrolebindings", "clusterrolebindings", desired.Name)
+
+	if err := controllerutil.SetControllerReference(controller, desired, r.Scheme); err != nil {
+		return fmt.Errorf("failed to set the controller reference for clusterrolebindings %s : %w", desired.Name, err)
+	}
+
+	exist, current, err := r.currentClusterRoleBinding(ctx, desired)
+	if err != nil {
+		return fmt.Errorf("failed to fetch current clusterrolebindings: %w", err)
+	}
+
+	if !exist {
+		if err := r.createClusterRoleBinding(ctx, desired); err != nil {
+			return err
+		}
+
+		_, current, err = r.currentClusterRoleBinding(ctx, desired)
+		if err != nil {
+			return fmt.Errorf("failed to fetch created clusterrolebindings: %w", err)
+		}
+		reqLogger.Info("created clusterrolebindings", "clusterrolebindings", desired.Name)
+	}
+
+	if err := r.updateClusterRoleBinding(ctx, current, desired); err != nil {
+		return fmt.Errorf("failed to update clusterrolebindings: %w", err)
+	}
+
+	return nil
+}
+
+// updateClusterRoleBindings updates the current clusterrolebindings and returns a flag to denote if the update was done.
+func (r *AWSLoadBalancerControllerReconciler) updateClusterRoleBinding(ctx context.Context, current *rbacv1.ClusterRoleBinding, desired *rbacv1.ClusterRoleBinding) error {
+	changed, err := isClusterRoleBindingChanged(current, desired)
+	if err != nil {
+		return err
+	}
+
+	if !changed {
+		return nil
+	}
+
+	updated := current.DeepCopy()
+	updated.Name = desired.Name
+	updated.Namespace = desired.Namespace
+	updated.Subjects = desired.Subjects
+	updated.RoleRef = desired.RoleRef
+
+	if err := r.Client.Update(ctx, updated); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func isClusterRoleBindingChanged(current *rbacv1.ClusterRoleBinding, desired *rbacv1.ClusterRoleBinding) (bool, error) {
```

`hasClusterRoleBindingChanged` sounds like a better name.

---

This function does not need to return an error. I don't see an error returned anywhere.

---

Good catch with the `error`, removing it across files.